src/rpc/core_rpc_server.cpp

```diff
@@ -1123,10 +1124,40 @@ namespace cryptonote
     if (use_bootstrap_daemon_if_necessary<COMMAND_RPC_GET_TRANSACTION_POOL_HASHES_BIN>(invoke_http_mode::JON, "/get_transaction_pool_hashes.bin", req, res, r))
       return r;
 
+    std::vector<crypto::hash> tx_pool_hashes;
     const bool restricted = m_restricted && ctx;
     const bool request_has_rpc_origin = ctx != NULL;
-    m_core.get_pool().get_transaction_hashes(res.tx_hashes, !request_has_rpc_origin || !restricted);
-    res.status = CORE_RPC_STATUS_OK;
+    m_core.get_pool().get_transaction_hashes(tx_pool_hashes, !request_has_rpc_origin || !restricted);
+
+    if (req.long_poll)
+    {
+      crypto::hash checksum = {};
+      for (crypto::hash const &hash : tx_pool_hashes) crypto::hash_xor(checksum, hash);
+
+      if (req.tx_pool_checksum == checksum)
+      {
+        time_t before = time(nullptr);
+        std::unique_lock<std::mutex> lock(m_core.m_long_poll_mutex);
+        if ((m_core.m_long_poll_connections + 1) > MAX_LONG_POLL_CONNECTIONS)
+        {
+          res.status = "Too many long polling connections, refusing connection";
+          return true;
+        }
+
+        m_core.m_long_poll_connections++;
+        std::cv_status status = m_core.m_long_poll_wake_up_clients.wait_for(lock, tools::wallet2::rpc_long_poll_timeout);
```

Comment:
This should be waiting on some condition, e.g. `_clients.wait_for(lock, tools::wallet2::rpc_long_poll_timeout, [&]() { ... })`

Not sure what the `...` should be: probably something like "number of transactions increased", both to block spurious wakeups, and to handle the case where the condition already became satisfied while acquiring the lock.